Spec/Auth.swift

```diff
@@ -637,9 +637,124 @@ class Auth : QuickSpec {
+                            expect(tokenDetails).toNot(beNil())
+                            expect(tokenDetails?.clientId) == testTokenDetails.clientId
+                            expect(tokenDetails?.capability) == testTokenDetails.capability
+                            expect(tokenDetails?.issued).toNot(beNil())
+                            expect(tokenDetails?.expires).toNot(beNil())
+                            if let issued = tokenDetails?.issued, let testIssued = testTokenDetails.issued {
+                                expect(issued.compare(testIssued)) == NSComparisonResult.OrderedSame
+                            }
+                            if let expires = tokenDetails?.expires, let testExpires = testTokenDetails.expires {
+                                expect(expires.compare(testExpires)) == NSComparisonResult.OrderedSame
+                            }
+                            done()
+                        })
+                    }
+                }
+
+                it("query will provide a TokenRequest") {
+                    var tokenRequest: ARTTokenRequest?
+                    waitUntil(timeout: testTimeout) { done in
+                        // Sandbox and valid TokenRequest
+                        ARTRest(options: AblyTests.commonAppSetup()).auth.createTokenRequest(nil, options: nil, callback: { newTokenRequest, error in
+                            expect(error).to(beNil())
+                            tokenRequest = newTokenRequest
+                            done()
+                        })
+                    }
+
+                    guard let testTokenRequest = tokenRequest else {
+                        fail("TokenRequest is empty")
+                        return
+                    }
+
+                    let encoder = ARTJsonLikeEncoder()
+                    encoder.delegate = ARTJsonEncoder()
+                    guard let jsonTokenRequest = encoder.encodeTokenRequest(testTokenRequest) else {
+                        fail("Invalid TokenRequest")
+                        return
+                    }
+
+                    let options = ARTClientOptions()
+                    options.authUrl = NSURL(string: "http://echo.ably.io")
+                    expect(options.authUrl).toNot(beNil())
+
+                    // JSON with TokenRequest
+                    options.authParams = [NSURLQueryItem]()
+                    options.authParams?.append(NSURLQueryItem(name: "type", value: "json"))
+                    options.authParams?.append(NSURLQueryItem(name: "body", value: jsonTokenRequest.toUTF8String))
+
+                    let rest = ARTRest(options: options)
+                    rest.httpExecutor = testHTTPExecutor
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.requestToken(nil, withOptions: nil, callback: { tokenDetails, error in
+                            expect(testHTTPExecutor.requests.last?.URL?.host).to(equal("echo.ably.io"))
```

Comment:
But I get your point and I'm doing the same for the other two (`TokenRequest` and `TokenDetails`).